pkg/user/service.go

```diff
@@ -134,3 +135,24 @@ func (s service) FindOrCreate(email string, password string) (*model.User, error
 func (s service) Delete(id uint) error {
 	return s.repository.delete(id)
 }
+
+func (s service) Update(id uint, email, password string) (*model.User, error) {
+	user, err := s.repository.findById(id)
+	if err != nil {
+		return nil, err
+	}
+
+	if email != "" {
+		user.Email = email
+	}
+
+	if password != "" {
+		var err error
+		user.Password, err = hashPassword(password)
+		if err != nil {
+			return nil, fmt.Errorf("password hashing failed: %s", err)
```

Comment:
nit: as mentioned in another comment we should settle on a standard phrase like `failed to ...`/`error ...`. Like `failed to hash password` or `error hashing password`

```suggestion
			return nil, fmt.Errorf("failed to hash password: %s", err)
```

Reply:
Also changed this in another place within the same file.